models/article-model.js

```diff
@@ -27,14 +26,36 @@ FROM
     articles
 LEFT JOIN 
     comments ON articles.article_id = comments.article_id
-GROUP BY 
-    articles.article_id
-ORDER BY 
-    articles.created_at DESC`
-      )
-      .then(({ rows }) => rows);
-    )
-    .then(({ rows }) => rows);
+    `;
+  const queryValues = [];
+
+  const promises = [];
+
+  if (topic) {
+    const topicExistsQuery = `SELECT EXISTS(SELECT 1 FROM topics WHERE slug = $1);`;
+    const topicExistsPromise = db.query(topicExistsQuery, [topic]);
+    promises.push(topicExistsPromise);
+    queryValues.push(topic);
+    query += ` WHERE topic = $1`;
+  }
+
+  query += ` GROUP BY articles.article_id 
+  ORDER BY articles.created_at DESC`;
+
+  const articlePromise = db.query(query, queryValues);
+
+  promises.push(articlePromise);
+  return Promise.all(promises).then((resolvedPromises) => {
+    if (topic) {
+      if (!resolvedPromises[0].rows[0].exists) {
+        return Promise.reject({ status: 404, msg: "not found" });

```

Comment:
Since this is a custom error handler, you can be a bit more precise and say 'Topic not found'